app/scripts/background.js

```diff
@@ -286,24 +286,27 @@ function maybeDetectPhishing(theController) {
 
       // Determine the block reason based on the type
       let blockReason;
+      let blockedUrl = hostname;
       if (phishingTestResponse?.result && blockedRequestResponse.result) {
         blockReason = `${phishingTestResponse.type} and ${blockedRequestResponse.type}`;
       } else if (phishingTestResponse?.result) {
         blockReason = phishingTestResponse.type;
       } else {
         blockReason = blockedRequestResponse.type;
+        blockedUrl = details.initiator;
       }
 
       theController.metaMetricsController.trackEvent({
         // should we differentiate between background redirection and content script redirection?
         event: MetaMetricsEventName.PhishingPageDisplayed,
         category: MetaMetricsEventCategory.Phishing,
         properties: {
-          url: hostname,
+          url: blockedUrl,
           referrer: {
-            url: hostname,
+            url: blockedUrl,
           },
           reason: blockReason,
+          requestDomain: blockedRequestResponse?.result ? hostname : undefined,
```

Comment:
Nit: Does this need the optional chaining? 🤔 Asking as above we have `if (phishingTestResponse?.result && blockedRequestResponse.result)`. Notice how the `blockedRequestResponse.result` does not have the optional chaining.